ArjunaJTA/jta/classes/com/arjuna/ats/jta/common/JTAEnvironmentBean.java

```diff
@@ -410,6 +411,11 @@ else if(instance != oldInstance)
      */
     public void setXaRecoveryNodes(List<String> xaRecoveryNodes)
     {
+        if(jtaLogger.logger.isDebugEnabled()) {
+            jtaLogger.logger.debugf("Setting up node identifiers '%s' for which recovery"
+                + " will be performed", xaRecoveryNodes);
```

Comment:
Does putting a List<String> here actually print out each element of the list?

Reply:
@mmusgrov it depends on the `toString()` method of the `List` implementation and the `ArrayList` is used here which really prints all the records. This is not true e.g. for arrays in java but from my experience it's true for all the collection java classes in the java util package.

Reply:
System.out.println("%s", arrayList) prints out each element as you suggest so I agree and would expect any logger should do the same.